intern.go

```diff
@@ -71,74 +70,109 @@ func Get(cmpVal interface{}) *Value {
 		v = valSafe[cmpVal]
 	} else if addr, ok := valMap[cmpVal]; ok {
 		v = (*Value)((unsafe.Pointer)(addr))
-	}
-	if v != nil {
-		v.refs++
+		// We've materialized a *Value out of thin air, in violation
+		// of Go's unsafety rules. The GC won't "see" this pointer
+		// until the next GC cycle. This is also true of normal
+		// pointers. Unlike normal pointers however, it's possible
+		// that the GC has just finished a mark phase in which it
+		// concluded that v was unreferenced, and thus collectable.
+		//
+		// By setting resurrected here, we signal to v's finalizer
+		// that the Value needs to remain alive for at least one more
+		// GC cycle, long enough for the GC to become aware of v's
+		// existence. See the comments in finalize for the other leg
+		// of this interaction.
+		v.resurrected = true
 		return v
 	}
-	v = &Value{cmpVal: cmpVal, refs: 1}
-	if valSafe != nil {
-		valSafe[cmpVal] = v
-	} else {
-		valMap[cmpVal] = uintptr(unsafe.Pointer(v))
-		runtime.SetFinalizer(v, finalize)
-	}
+	// When we first create a value, v is a "correct" pointer from the
+	// runtime's perspective. There is no risk of concurrent
+	// collection, so we can set resurrected=false.
+	v = &Value{cmpVal: cmpVal, resurrected: false}
```

Comment:
let's not set zero values.

The resurrected field was documented in my change ("// It is set true whenever v is synthesized from a uintptr.") so it's clear already from those fields why it doesn't need to be set.